controllers/movement.py

```diff
@@ -46,56 +46,36 @@ def __init__(self) -> None:
         # When on True, a trajectory is genearted every code run and will consistently
         # be painted onto Glass' SmartDashboard.
         # When on False, a trajectory is only generated when needed to save robotRIO resources.
-        self.debug_trajectory = False
+        self.debug_trajectory = tunable(False)
 
-        # Initialise the trajectory config and add constraints.
-        # These will be used until set_goal() is called, which will then deifne a new
-        # config and constraints.
-        self.config = TrajectoryConfig(1, 1.5)
-        self.config.addConstraint(CentripetalAccelerationConstraint(1.5))
-        topRight = Translation2d(self.goal.X() + 2, self.goal.Y() - 2)
-        bottomLeft = Translation2d(self.goal.X() - 2, self.goal.Y() + 2)
-        self.config.addConstraint(
-            RectangularRegionConstraint(bottomLeft, topRight, MaxVelocityConstraint(1))
-        )
+        self.set_goal(Pose2d(3, 0, 0), Rotation2d(0))
 
     def setup(self):
         self.robot_object = self.field.getObject("auto_trajectory")
 
     def generate_trajectory(self) -> Trajectory:
-        # Grab all required information from self.chassis so we don't repeately run
-        # getters.
-        self.chassis_velocity = self.chassis.get_velocity()
-        self.x_pos = self.chassis.get_pose().X()
-        self.y_pos = self.chassis.get_pose().Y()
-        self.x_velocity = self.chassis_velocity.vx
-        self.y_velocity = self.chassis_velocity.vy
-
-        x_translation = self.goal.X() - self.x_pos
-        y_translation = self.goal.Y() - self.y_pos
-        translation_distance = (
-            0.00  # Pre-define translation_distance so it can never be Unbound.
-        )
-        translation_distance = math.sqrt(
-            x_translation**2 + y_translation**2
-        )  # Calculate the distance between the robot and the goal to be used later.
+        chassis_velocity = self.chassis.get_velocity()
+        x_pos = self.chassis.get_pose().X()
+        y_pos = self.chassis.get_pose().Y()
+        x_velocity = chassis_velocity.vx
+        y_velocity = chassis_velocity.vy
+
+        translation = self.goal.translation() - Pose2d(x_pos, y_pos, 0).translation()
```

Comment:
why construct a new pose to then immediately get its translation and throw it away? don't you already have a pose with this information?